dev/tools/gen_keycodes/data/keyboard_key.tmpl

```diff
@@ -148,36 +145,7 @@ class LogicalKeyboardKey extends KeyboardKey {
 
   /// The debug string to print for this keyboard key, which will be null in
   /// release mode.
-  final String? debugName;
-
-  /// The Unicode string representing the character produced by a [RawKeyEvent].
-  ///
-  /// This value is useful for describing or matching mnemonic keyboard
-  /// shortcuts.
-  ///
-  /// This value is an empty string if there's no key label data for a key.
-  ///
-  /// On most platforms this is a single code point, but it could contain any
-  /// Unicode string. The `keyLabel` differs from [RawKeyEvent.character]
-  /// because `keyLabel` only takes into account the key being pressed, not any
-  /// combining keys pressed before it, so, for example, an “o” that follows a
-  /// combining dieresis (“¨”, COMBINING DIAERESIS (U+0308)) would just return
-  /// “o” for [keyLabel], but would return “ö” for [RawKeyEvent.character].
-  ///
-  /// {@macro flutter.services.RawKeyEventData.keyLabel}
-  final String keyLabel;
-
-  @override
-  int get hashCode => keyId.hashCode;
-
-  @override
-  bool operator ==(Object other) {
-    if (other.runtimeType != runtimeType) {
-      return false;
-    }
-    return other is LogicalKeyboardKey
-        && other.keyId == keyId;
-  }
+  String? get debugName => _debugNames[keyId];
```

Comment:
Couldn't this also include a representation of the key as a Unicode character, if it is part of the Unicode ID plane?

Reply:
Are you saying to include it as the release information or the debug information?
- If as release information, isn't it redundant symbol since it's unnecessary for runtime logic?
- If as debug information, doesn't the current `debugName` already convey the key name?

Reply:
As debug information only.  It would include a lot more keys that aren't in your `_debugNames` map: any Unicode-producing key would have a `debugName` that includes the character it represents. You could just produce a string if it doesn't exist in the map. If you do that here, then it'll also get printed as part of the `debugFillProperties` output.

Reply:
I've tried to add a `debugCharacterLabel` as you describe. See https://github.com/dkwingsmt/flutter/commit/c62cf0803fe682b02e9586cf7937c0e9f3fb5e80

It doesn't really work well in my opinion.
- As you said, this is _debug_ information only. Therefore it doesn't solve the problem mentioned in https://github.com/flutter/flutter/pull/78263#discussion_r595331249, which asks to add more information to _release mode_ code.
- The only "unicode" keys `LogicalKeyboardKey` represents are ASCII printable keys (just search for `0x000` in `keyboard_key.dart`). There isn't a `œ` logical key: despite that this chacacter can be input by a macOS keyboard with Option-Q, it's still an Option key followed by a KeyQ key. The `debugCharacterLabel` does not add any information besides what's already available in `debugName`.

Reply:
I was thinking of something like this:

```dart
 String? get debugName {
    String? result;
    assert(() {
	  result ??= _debugNames[keyId];
      if (keyId & ~valueMask == unicodePlane) {
        result = 'Key ${String.fromCharCode(keyId)}';
      }
      return true;
    }());
    return result;
  }
```

For the printable key with the key cap of "Ù" (letter U with grave) on the French keyboard, the function above will return 'Key Ù', but your current implementation would return `null`.  It's more than just ASCII: any key that generates a Unicode character should get a Unicode plane key id. In this case the logical id for "Ù" is 0x000000D9. There is a dedicated key for this on the French keyboard, so that's exactly what it should report. Debugging will get a lot harder if all they get from `debugName` is `null` outside of the ASCII keys, and this adds more useful information to allow them to differentiate inputs.

Reply:
Actually, this might be even better, since it won't return null except in release mode.

```dart
 String? get debugName {
    String? result;
    assert(() {
	  result ??= _debugNames[keyId];
      if (result == null) {
        if (keyId & ~valueMask == unicodePlane) {
          result = 'Key ${String.fromCharCode(keyId)}';
        } else {
          result = 'Key with ID 0x${keyId.toRadixString(16).padLeft(digits, '0')}';
        }
      }
      return true;
    }());
    return result;
  }
```

Reply:
Thanks for the advice. I've added this one, as well as similar code to PhysicalKeyboardKey, and tests.

Reply:
It's weird: I see your push with the `debugName` changes in it (https://github.com/flutter/flutter/pull/78263/commits/0deb173690c8c162d49bafa5bdd11bd25613957c), but I don't see them here in the "Files changed" tab on GitHub.  Maybe you somehow reverted them?